service/controller/appcatalog/v1/resource/index/desired_test.go

```diff
@@ -0,0 +1,137 @@
+package index
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_Resource_GetDesiredState(t *testing.T) {
+	tests := []struct {
+		name              string
+		obj               *v1alpha1.AppCatalog
+		expectedConfigMap *corev1.ConfigMap
+		h                 func(w http.ResponseWriter, r *http.Request)
+		errorMatcher      func(error) bool
+	}{
+		{
+			name: "case 0: flawless flow",
+			obj: &v1alpha1.AppCatalog{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "giantswarm",
+					Namespace: "default",
```

Comment:
Lets remove the namespace and labels from the test. The CR is cluster scoped so setting a namespace doesn't make sense.